NuRadioReco/modules/phasedarray/triggerSimulator.py

```diff
@@ -374,66 +394,51 @@ def run(self, evt, station, det,
-                                                                                       window_time, cut_times,
-                                                                                       trigger_adc=trigger_adc,
-                                                                                       upsampling_factor=upsampling_factor,
-                                                                                       nyquist_zone=nyquist_zone,
-                                                                                       bandwidth_edge=bandwidth_edge)
-            else:
-                primary_trigger, trigger_delays, dummy_delays = self.phased_trigger(station, det, beam_rolls,
-                                                                                    empty_rolls,
-                                                                                    triggered_channels, threshold,
-                                                                                    window_time, cut_times,
-                                                                                    trigger_adc=trigger_adc,
-                                                                                    upsampling_factor=upsampling_factor,
-                                                                                    nyquist_zone=nyquist_zone,
-                                                                                    bandwidth_edge=bandwidth_edge)
-                secondary_trigger, sec_trigger_delays, dummy_delays = self.phased_trigger(station, det,
-                                                                                          secondary_beam_rolls,
-                                                                                          empty_rolls,
-                                                                                          secondary_channels, threshold,
-                                                                                          window_time, cut_times,
-                                                                                          trigger_adc=trigger_adc,
-                                                                                          upsampling_factor=upsampling_factor,
-                                                                                          nyquist_zone=nyquist_zone,
-                                                                                          bandwidth_edge=bandwidth_edge)
+            if ((not coupled) and (len(secondary_channels) > 0)):
+                primary_trigger, trigger_delays, dummy_delay = phased_trigger(station, det, threshold,
+                                                                              triggered_channels,
+                                                                              phasing_angles,
+                                                                              [],
```

Comment:
Imo for empty arguments, e.g. [], better to name the parameter for clarity, e.g. "secondary_channels=[]" That or re-work the phased_trigger() function so the arguments are truly optional and then just omit them?